lib/ssh_crypto_policy.pm

```diff
@@ -0,0 +1,109 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+# Summary: Package for ssh cryptographic policy testing
+#
+# Maintainer: George Gkioulis <ggkioulis@suse.com>
+
+package ssh_crypto_policy;
+use base 'consoletest';
+use testapi;
+use utils;
+use strict;
+use warnings;
+
+sub new() {
+    my ($class, %args) = @_;
+    my $self = $class->SUPER::new();
+    $self->{name}            = $args{name};
+    $self->{query}           = $args{query};
+    $self->{cmd_option}      = $args{cmd_option};
+    $self->{algorithm_array} = ();
+
+    # In the case of HostKeyAlgorithms, get only the ones provided by the server side
+    if ($self->{name} eq "HostKeyAlgorithms") {
+        $self->create_host_key_algorithm_array();
+    } else {
+        # Split the output of the ssh algorithm query to an array
+        $self->{algorithm_array} = [split(/\n/, script_output("ssh -Q $self->{query}"))];
+    }
+
+    return $self;
+}
+
+sub create_host_key_algorithm_array() {
+    my ($self) = @_;
+
+    # If nmap is not installed, install it
+    if (script_run("which nmap")) {
+        zypper_call("in nmap");
+    }
+
+    # Get all the algorithms supported by the server side
+    my $output       = script_output("nmap --script ssh2-enum-algos -sV -p 22 localhost");
+    my @output_lines = split('\|', $output);
+
+    my $parse_algorithms = 0;
+    foreach my $line (@output_lines) {
+        if ($parse_algorithms) {
+            # If line contains the string "encryption algorithms", you can finish parsing
+            if (index($line, "encryption_algorithms") != -1) {
+                last;
+            } else {
+                # Get the host key algorithms
+                $line =~ s/^\s+|\s+$//g;
+                push(@{$self->{algorithm_array}}, $line);
+            }
```

Comment:
```suggestion
            last if (index($line, "encryption_algorithms") != -1) 
            # Get the host key algorithms
            $line =~ s/^\s+|\s+$//g;
            push(@{$self->{algorithm_array}}, $line);
```
More readable?

Reply:
@ge0r will this work please?

Reply:
For sure it is more elegant, it should work.

Reply:
For sure it is more elegant, it should work

Reply:
Just make sure there is a `;` at the end, as in `last if (index($line, "encryption_algorithms") != -1);`